subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/RepoScriptBlockUtil.groovy

```diff
@@ -73,9 +74,14 @@ class RepoScriptBlockUtil {
 
         private MirroredRepository(String originalUrl, String mirrorUrl, String type) {
             this.originalUrl = originalUrl
-            this.mirrorUrl = mirrorUrl ?: originalUrl
             this.name = mirrorUrl ? name() + "_MIRROR" : name()
             this.type = type
+            if (OperatingSystem.current().isMacOsX()) {
+                // https://github.com/gradle/gradle-private/issues/2725
+                this.mirrorUrl = originalUrl
```

Comment:
Why do we need to do this here as well? Isn't the configuration enough?

Reply:
You're right. Removed.

Reply:
Wait.... the Kotlin DSL configuration only changes `org.gradle.internal.plugins.portal.url.override`, but here the configurations are passed via `REPO_MIRROR_URLS` env variables: https://github.com/gradle/gradle/blob/f604a9b7daa45b7475aae60786a0870be983d26e/buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt#L249

So this is still necessary...

Reply:
How about we override the env variable on macOS builds via TC configuration?

Reply:
Sounds good.